code/to_bsdd.py

```diff
@@ -171,28 +172,32 @@ def get_name(n):
 def reduce_description(s, trim=True):
     """ Remove all the unnecesarry characters, strip <html sections>, and trim the description to bare minimum. """
     try:
-        s1 = html.unescape(s)
+        s = html.unescape(s)
     except TypeError:
         # error when name contains link, for example: "https://github.com/buildingSMART/IFC4.3.x-development/edit/master/docs/schemas/core/IfcProductExtension/Types/IfcAlignmentTypeEnum.md#L0 has no content"
-        s1 = ''
+        s = ''
     if trim:
-        # split and remove on keywords:
-        i = min([x for x in [s1.find('NOTE'), s1.find('DIAGRAM'), s1.find('CHANGE'), s1.find('IFC4'), s1.find('HISTORY'), s1.find('REFERENCE'), s1.find('EXAMPLE'), s1.find('DEPRECATION'), 1e5] if x >= 0])
+        # split and remove on keywords:        
+        j = re.search(r'\$\$', s).start() if re.search(r'\$\$', s) else -1
+        k = re.search(r'(Base|General) formula', s).start() if re.search(r'(Base|General) formula', s) else -1
+        i = min([x for x in [s.find('NOTE'), s.find('DIAGRAM'), s.find('CHANGE'), s.find('IFC4'), s.find('HISTORY'), s.find('REFERENCE'), s.find('EXAMPLE'), s.find('DEPRECATION'), j, k, 1e5] if x >= 0])
         if i > 0 and i != 1e5:
-            s1 = s1[:i]
+            s = s[:i]
         elif i==0:
             # some descriptions start with History or Note. Don't want them.
-            s1 = ""
-    s2 = re.sub('\n', '; ', s1)
-    s3 = re.sub(':', ': ', s2)
-    s4 = re.sub(HTML_TAG_PATTERN, ' ', s3)
-    s5 = re.sub(CURLY_BRACKET_PATTERN, ' ', s4)
-    s6 = re.sub(r'SELF\\', '', s5)
-    s7 = re.sub(r"\s+", " ", s6)
-    s8 = re.sub(FIGURE_PATTERN, '', s7)
-    s9 = re.sub(LIST_PATTERN, '', s8)
-    s10 = re.sub(MULTIPLE_SPACE_PATTERN, ' ', s9).strip()
-    return clean(s10)
+            s = ""
+    s = re.sub('[*]{2}', ' ', s)    # remove double asteriks (used in formulas and for text in bold
```

Comment:
The more idiomatic way to write this would be:

```suggestion
    s = re.sub(r'\*{2}', ' ', s)    # remove double asteriks (used in formulas and for text in bold
```

i.e the single character `*`, but escaped because it has a special meaning in regex.